src/sendalert.cpp

```diff
@@ -71,14 +71,14 @@ void ThreadSendAlert()
     //
     CAlert alert;
     alert.nRelayUntil   = GetTime() + 15 * 60;
```

Comment:
Does this mean the alert will be relayed for just 15 minutes, so that anyone who start up their old node _after_ June 25 will silently fork away?

Reply:
```
    //
    // Alerts are relayed around the network until nRelayUntil, flood
    // filling to every node.
    // After the relay time is past, new nodes are told about alerts
    // when they connect to peers, until either nExpiration or
    // the alert is cancelled by a newer alert.
    // Nodes never save alerts to disk, they are in-memory-only.
    //
```
So someone starting up an old node after June 25, let's say they have an old peers.dat file with ip addresses of some other old nodes which are still online, when they connect they will receive the alert and go into safe mode.

Reply:
Ah, so it keeps relaying even after `nRelayUntil`, but only by "pull" when nodes connect rather than by "push" to currently connected nodes? Yeah that's fine.

Why do you assume that "they have an old peers.dat file with ip addresses of some other old nodes which are still online"? That's probably false: most of the Overwinter-incompatible nodes will be taken offline by their operator soon after they receive the alert (because they'll become mostly useless).

Reply:
There might be some headless nodes out there, on a static IP, which have been left to run mostly unattended.